src/test/java/codeu/model/store/basic/UserStoreTest.java

```diff
@@ -0,0 +1,92 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.User;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class UserStoreTest {
+
+  private UserStore userStore;
+  private PersistentStorageAgent mockPersistentStorageAgent;
+
+  private final User USER_ONE =
+      new User(UUID.randomUUID(), "test_username_one", Instant.ofEpochMilli(1000));
```

Comment:
This is actually a suggested change on the User class; perhaps the constructor should be changed to generate the UUID, rather than passing one in all the time. Similarly, the constructor could also generate the timestamp. I believe a constructor with inputs is needed for loading objects from storage, but in general the inputs are not needed (IMO).

Reply:
If we make this change, we should probably make it for all of the data classes, and I'd rather do that in a follow-up PR. We could also go more extreme and make the data classes into beans, which means they'd have a no-args constructor and a bunch of setter functions.

One benefit of the current approach is that it makes it very obvious where you need to change stuff if you add a field to a data object. For example if you add a password to the `User` class, you know you have to add it to the constructor, and then you'll get a compiler error in all of the places you need to change it. If we use a "smart" constructor it might be tempting to do something like generate a password inside the constructor, which will make it less obvious where changes need to happen.

I'd lean towards leaving things as-is, unless you feel strongly. (Or unless you want to take it on :stuck_out_tongue: )

Reply:
Definitely a follow-on item. I think there's a middle ground where you do common work within the constructor without making it a bean. UUID is a bit different than authentication/verification methods. UUID is a fundamental property of the object in our system, and we shouldn't rely on the caller of the constructor to do the right thing. Testing is one exception to this, as is instantiation by another trusted entity such as the persistent storage module.

Reply:
Agreed. Let's leave this as a follow-up. Plenty of other stuff to do in the meantime.